### PR TITLE
vtol tailsitter attitude control

### PR DESCRIPTION
**Pull request by jingego**
Revised transition start pitch angle to current vehicle's attitude instead of setpoint.
Fixed roll control in fixed-wing frame during transition.

[Flight Log](https://www.dropbox.com/s/3f8y0uadqdyjly8/02_59_44.ulg?dl=0)

### CODE REVIEW

File: src/modules/vtol_att_control/tailsitter.cpp [hunk 1]
```diff
@@ -236,7 +237,10 @@ void Tailsitter::update_transition_state()
 	if (!_flag_was_in_trans_mode) {
 		// save desired heading for transition and last thrust value
 		_yaw_transition = _v_att_sp->yaw_body;
-		_pitch_transition_start = _v_att_sp->pitch_body;
+		//transition should start from current attitude instead of current setpoint
+		math::Quaternion q_att(_v_att->q);
```
Comment (dagar) [inline src/modules/vtol_att_control/tailsitter.cpp]: You can use the newer matrix math library to do this, which is already being used in tailsitter.
https://github.com/PX4/Firmware/blob/master/src/modules/vtol_att_control/tailsitter.cpp#L131

**Comment by dagar** [comment]
Looks good. Have you tried testing tailsitter in Gazebo with SITL? https://dev.px4.io/en/simulation/gazebo.html

You can upload your log to https://logs.px4.io/ for easy review.
https://logs.px4.io/plot_app?log=54638b9c-4724-42a0-b722-9c163b1d8ac8

**Comment by jingego** [comment]
we tested it on gazebo and then conducted several flight tests.

File: src/modules/vtol_att_control/tailsitter.cpp (hunk 1 above)
Comment (jingego) [inline src/modules/vtol_att_control/tailsitter.cpp]: thx! i will revise it.

**Pull request merged**